test/integration/src/test/java/io/pravega/test/integration/BoundedStreamReaderTest.java

```diff
@@ -230,6 +250,128 @@ public void testBoundedStreamWithScaleTest() throws Exception {
+        writer1.writeEvent(keyGenerator.apply("0.4"), getEventData.apply(0)).get();
+        //2. Write 2 events to S1.
+        writer1.writeEvent(keyGenerator.apply("0.7"), getEventData.apply(1)).get();
+        writer1.writeEvent(keyGenerator.apply("0.7"), getEventData.apply(1)).get();
+
+        //2.Scale stream
+        Map<Double, Double> newKeyRanges = new HashMap<>();
+        newKeyRanges.put(0.0, 0.6);
+        newKeyRanges.put(0.6, 1.0);
+        scaleStream(STREAM1, newKeyRanges);
+
+        //3.Write 2 events to Segment S2 [0.0, 0.6)
+        writer1.writeEvent(keyGenerator.apply("0.3"), getEventData.apply(2)).get();
+        writer1.writeEvent(keyGenerator.apply("0.3"), getEventData.apply(2)).get();
+        writer1.writeEvent(keyGenerator.apply("0.3"), getEventData.apply(2)).get();
+        //4.Write 2 events to Segment S3 [0.6, 1.0)
+        writer1.writeEvent(keyGenerator.apply("0.7"), getEventData.apply(3)).get();
+        writer1.writeEvent(keyGenerator.apply("0.7"), getEventData.apply(3)).get();
+        writer1.writeEvent(keyGenerator.apply("0.7"), getEventData.apply(3)).get();
+
+        @Cleanup
+        ReaderGroupManager groupManager = ReaderGroupManager.withScope(SCOPE, controllerUri);
+
+        ReaderGroupConfig readerGroupCfg1 = ReaderGroupConfig.builder().disableAutomaticCheckpoints()
+                                                             .stream(Stream.of(SCOPE, STREAM1))
+                                                             .build();
+        groupManager.createReaderGroup("group", readerGroupCfg1);
+        ReaderGroup readerGroup = groupManager.getReaderGroup("group");
+
+        //Create two readers
+        @Cleanup
+        EventStreamReader<String> reader1 = clientFactory.createReader("readerId1", "group", serializer,
+                                                                       ReaderConfig.builder().build());
+        @Cleanup
+        EventStreamReader<String> reader2 = clientFactory.createReader("readerId2", "group", serializer,
+                                                                       ReaderConfig.builder().build());
+
+        //5. Read 1 event on both the readers.
+        String event4Reader1 = reader1.readNextEvent(10000).getEvent();
+        String event4Reader2 = reader2.readNextEvent(10000).getEvent();
+        assertTrue(event4Reader1 != null);
+        assertTrue(event4Reader2 != null);
+
+        //6. Continue reading on the reader where event data 1 is observed.
+        if (event4Reader1.equals(getEventData.apply(1))) {
+            assertEquals(getEventData.apply(1), reader1.readNextEvent(10000).getEvent());
+            EventRead<String> stringEventRead = reader1.readNextEvent(10000);
+            assertEquals(getEventData.apply(3), stringEventRead.getEvent());
+            reader1.close();
+            readerGroup.readerOffline("readerId1", stringEventRead.getPosition());
```

Comment:
This should be redundant with the call to close right above.

Reply:
corrected the code.